subsys/fs/nvs/nvs_priv.h

```diff
@@ -30,11 +41,18 @@ extern "C" {
 
 /* Allocation Table Entry */
 struct nvs_ate {
-	uint16_t id;	/* data id */
-	uint16_t offset;	/* data offset within sector */
-	uint16_t len;	/* data len within sector */
-	uint8_t part;	/* part of a multipart data - future extension */
-	uint8_t crc8;	/* crc8 check of the entry */
+#if NVS_OFFSET_BITS > 16
+	/* Order to keep proper alignment within struct */
+	uint32_t offset;   /* data offset within sector */
+	uint16_t len;      /* data length */
+	uint16_t id;       /* data id */
+#else
+	uint16_t id;       /* data id */
```

Comment:
Is this to optimise things for 16-bit microcontrollers or to save data space?

Reply:
Primarily compatibility reasons. The ATEs are stored on the flash. Changing their format would brake an existing partition created with an older version.